Make layer naming more consistent and indicative

mikini opened the pull request:
* Use "Transport" for both layers showing public transport cartographies
* Indicate cartography name of transport layers in parenthesis to distinguish
* Also eliminate "map" in layer names, which seems redundant

This is a follow-up to a comment regarding naming on blog post
https://blog.openstreetmap.org/2020/07/10/opnvkarte-a-new-featured-layer-on-www-openstreetmap-org/
(still unmoderated as of this commit).

gravitystorm commented:
See also https://github.com/openstreetmap/openstreetmap-website/pull/2670#issuecomment-650037548 where I describe what I think the naming conventions should be based on.

I should note here that "TF" is not a good contraction of Thunderforest, since Thunderforest is just one word and the f is never capitalised. 

Also, ÖPNV is neither the name of the map style (ÖPNVKarte), nor of the provider (MeMoMaps), so I think it's worth changing that to something else too.

mmd-osm commented:
So what is the final conclusion here? Adjust the pull request or maybe just close it?

mikini commented:
I have rebased the branch and added a new commit reflecting the discussions here and in #2670 as pr. my understanding of consensus. Space for longer texts were added by simply decreasing the font size of the layer name label.

The selector now looks like below (screenshot is of browser manipulated production site, did not succeed yet in getting dev environment running to test the code, so might have missed something);

![Screenshot from 2023-01-13 01-58-41](https://user-images.githubusercontent.com/1330166/212220181-d89e957f-633b-4699-bd6a-f6eb84d89dd7.png)

mmd-osm commented:
Some recent changes to the layer list: ÖPNVKarte was removed and Tracestrack Topo added:

![image](https://github.com/openstreetmap/openstreetmap-website/assets/5842757/becfd272-8049-4c27-baaf-a02d8acae6d3)
